fe/fe-core/src/main/java/org/apache/doris/planner/DistributedPlanner.java

```diff
@@ -557,6 +557,12 @@ private boolean dataDistributionMatchEqPredicate(List<BinaryPredicate> eqJoinPre
                 cannotReason.add(DistributedPlanColocateRule.COLOCATE_GROUP_IS_NOT_STABLE);
                 return false;
             }
+        } else {
+            // query same table but hit different rollup, colocate join should be disable
+            if (leftRoot.getSelectedIndexId() != rightRoot.getSelectedIndexId()) {
+                cannotReason.add(DistributedPlanColocateRule.QUERY_SAME_TABLE_BUT_HIT_DIFF_ROLLUP);
+                return false;
```

Comment:
if table is a colocate table, it could do colocate join with itself when group is stable